crypto/src/dex/swap.rs

```diff
@@ -1,33 +1,101 @@
 mod ciphertext;
 mod plaintext;
+
+use anyhow::Result;
+use ark_ff::PrimeField;
 pub use ciphertext::SwapCiphertext;
+use decaf377::{Element, FieldExt, Fq};
+use decaf377_ka::Public;
```

Comment:
nit: probably better not to interleave the re-exports defining the exposed api with the module's own use statements